src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/caseworker/ResendNewPinCallbackHandler.java

```diff
@@ -0,0 +1,138 @@
+    public ResendNewPinCallbackHandler(
+        CaseDetailsConverter caseDetailsConverter,
+        UserService userService,
+        CaseMapper caseMapper,
+        ClaimIssuedNotificationService claimIssuedNotificationService,
+        NotificationsProperties notificationsProperties
+    ) {
+        this.caseDetailsConverter = caseDetailsConverter;
+        this.userService = userService;
+        this.caseMapper = caseMapper;
+        this.claimIssuedNotificationService = claimIssuedNotificationService;
+        this.notificationsProperties = notificationsProperties;
+    }
+
+    @Override
+    protected Map<CallbackType, Callback> callbacks() {
+        return ImmutableMap.of(
+            CallbackType.ABOUT_TO_SUBMIT, this::resendNewPin
+        );
+    }
+
+    @Override
+    public List<CaseEvent> handledEvents() {
+        return EVENTS;
+    }
+
+    @Override
+    public List<Role> getSupportedRoles() {
+        return ROLES;
+    }
+
+    private CallbackResponse resendNewPin(CallbackParams callbackParams) {
+        Claim claim = caseDetailsConverter.extractClaim(callbackParams.getRequest().getCaseDetails());
+        String authorisation = callbackParams.getParams().get(CallbackParams.Params.BEARER_TOKEN).toString();
+
+        if (getDefendantLinkStatus(claim).isLinked()) {
+            logger.info("Claim {} has already been linked to defendant - cannot send notification",
+                claim.getExternalId());
+
+            return AboutToStartOrSubmitCallbackResponse.builder()
+                .errors(ImmutableList.of("Claim has already been linked to defendant - cannot send notification"))
+                .build();
+        }
+
+        if (!claim.getClaimData().getDefendant().getEmail().isPresent()) {
+            logger.info("Claim {} doesn't have defendant email address - cannot send notification",
+                claim.getExternalId());
+
+            return AboutToStartOrSubmitCallbackResponse.builder()
+                .errors(ImmutableList.of("Claim doesn't have defendant email address - cannot send notification"))
+                .build();
+        }
+
+        GeneratePinResponse pinResponse = getPinResponse(claim.getClaimData(), authorisation)
+            .orElseThrow(() -> new IllegalArgumentException("Pin generation failed"));
+
+        claim.getClaimData().getDefendant().getEmail().ifPresent(defendantEmail ->
+            claimIssuedNotificationService.sendMail(
+                claim,
+                defendantEmail,
+                pinResponse.getPin(),
+                notificationsProperties.getTemplates().getEmail().getDefendantClaimIssued(),
+                "defendant-issue-notification-" + claim.getReferenceNumber(),
+                claim.getClaimData().getDefendant().getName()
+            ));
+
+        Claim updatedClaim = claim.toBuilder()
+            .letterHolderId(pinResponse.getUserId())
+            .build();
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
```

Comment:
Sending the email with new PIN before persisting in DB is risky. We could send the email but fail to store the PIN meaning the user would get an email with an incorrect PIN. It would be best to persist first and then send the email once this is successful.

Not sure if this is possible with CCD though. Just raising this as a risk.

Reply:
@dharmendrak what do you think about this? should i move sending email to Submiited callback?

Reply:
Ignore this bit, this won't work as Pin is not stored in CCD to get picked in another callback